Source/ASImageNode.mm

```diff
@@ -756,17 +756,16 @@ - (void)asyncTraitCollectionDidChangeWithPreviousTraitCollection:(ASPrimitiveTra
   [super asyncTraitCollectionDidChangeWithPreviousTraitCollection:previousTraitCollection];
 
   if (AS_AVAILABLE_IOS_TVOS(12, 10)) {
-    __instanceLock__.lock();
+    AS::MutexLocker l(__instanceLock__);
       // update image if userInterfaceStyle was changed (dark mode)
-      if (_image != nil && self.primitiveTraitCollection.userInterfaceStyle != previousTraitCollection.userInterfaceStyle) {
-          UITraitCollection *tc = [UITraitCollection traitCollectionWithUserInterfaceStyle:self.primitiveTraitCollection.userInterfaceStyle];
+      if (_image != nil && _primitiveTraitCollection.userInterfaceStyle != previousTraitCollection.userInterfaceStyle) {
+          UITraitCollection *tc = [UITraitCollection traitCollectionWithUserInterfaceStyle:_primitiveTraitCollection.userInterfaceStyle];
           // get an updated image from asset catalog
-          UIImage *updatedImage = [self.image.imageAsset imageWithTraitCollection:tc];
+          UIImage *updatedImage = [_image.imageAsset imageWithTraitCollection:tc];
```

Comment:
Should we use the new method in iOS13+? Seems to have more sophisticated logic around loading image variants

https://developer.apple.com/documentation/uikit/uiimage/3294226-imagenamed?language=objc

Reply:
we can't use it in `ASImageNode` cause image is passed directly

Reply:
I think that may be good for a follow up; This PR is just about using the C++ / macro lock and then accessing ivars directly. Thoughts?

Reply:
@vovasty is right; we already have a `UIImage` in this method.